modules/services/window-managers/xmonad.nix

```
@@ -75,27 +75,57 @@ in {
           by Home Manager.
         '';
       };
+
+      libFiles = mkOption {
+        type = types.attrs;
```

```suggestion
        type = types.attrsOf (types.oneOf [ types.package types.path ]);
```

More precise tying.

Would it make sense to allow the user to use a multiline string directly here?
In which case the `oneOf` should be changed to `oneOf [ ..... types.lines ]`.

---

The values in `libFiles` are (right now at least, see rycee's comment below) assigned to `home.file.".xmonad/lib/Something.hs".source` which means, if I understand the documentation correctly, that only `path`-values are allowed. So maybe the type should be `types.attrsOf types.path;` On the other hand: it would of course be nice to be able to use `lines` also (as the name of the files is anyway given in the key).